### PR TITLE
g.list: Add test file

### PR DESCRIPTION
**Pull request by NishantBansal2003**
ref: #5848 

This PR adds a regression test suite for the `g.list` module. These tests are intended to serve as regression checks for upcoming PRs that add JSON support to `g.list`.

### CODE REVIEW

**Comment by NishantBansal2003** [comment]
I’m not sure why the output varies for the `test_verbose_listing_output` function (`-f` flag) on the OSGeo4W CI (see: https://github.com/OSGeo/grass/actions/runs/15714853038/job/44281933743#step:12:1341.
Also not sure why the `test_default_output` test returns a non-zero exit code.

**Comment by petrasovaa** [comment]
That is strange indeed, looks like 2 different issues. Could you test using just a single mapset, not switch them?

**Comment by NishantBansal2003** [comment]
> That is strange indeed, looks like 2 different issues. Could you test using just a single mapset, not switch them?

It looks like the same issue occurs with a single mapset, even without switching.

File: general/g.list/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import os
+import grass.script as gs
+import pytest
+
+
+@pytest.fixture
```
Comment (petrasovaa) [inline general/g.list/tests/conftest.py:8]:
```suggestion
@pytest.fixture(scope="module")
```

**Comment by petrasovaa** [comment]
I added the module scope to the fixture, but that behaves completely opposite than I expected, I thought it should not be executed for every function. Am I missing something? Looking at v.fill.holes test, it looks very similar, I don't see why this is failing...

@echoix any idea what to test?

**Comment by echoix** [comment]
If you not set it to once per module, does it only fail for the first test, and fine for the other ones? If so, it's #4480

**Comment by petrasovaa** [comment]
#4880 seems unrelated, typo perhaps?

**Comment by petrasovaa** [comment]
But yes, that's the behavior.

**Comment by echoix** [comment]
> #4880 seems unrelated, typo perhaps?

#4480, sorry I edited above

**Comment by echoix** [comment]
The first call to a C library tool from Python on windows will fail the test, but will have done something special somewhere that makes the following work. When doing with pytest-xdist (multiple workers), it's the first usage of a c library tool per worker that will fail.

It is often hidden because of xfail for another part of the test, and that we always run in the same order. Shiffling the order with a plugin makes this very obvious

**Comment by echoix** [comment]
To workaround, before waiting that I explain what I know so far in #4480, use the workaround I used extensively in not submitted PRs, and in https://github.com/OSGeo/grass/pull/4784, plus #5349 for having monkeypatch in another scope. Also similar to https://github.com/echoix/grass/pull/380

**Comment by NishantBansal2003** [comment]
OK, so the `test_default_output` can be fixed, but there’s another issue: on Windows CI, the `test_verbose_listing_output` function produces different output with the `-f` flag. See: https://github.com/OSGeo/grass/actions/runs/15714853038/job/44281933743#step:12:1341

**Comment by petrasovaa** [comment]
> OK, so the `test_default_output` can be fixed, but there’s another issue: on Windows CI, the `test_verbose_listing_output` function produces different output with the `-f` flag. See: https://github.com/OSGeo/grass/actions/runs/15714853038/job/44281933743#step:12:1341

Right, I think we would have to see if the title is written (check with r.info) and the problem is in reading it. But it's not terribly important feature, so I am fine with marking this as failing on Windows and move on.

**Comment by NishantBansal2003** [comment]
> Right, I think we would have to see if the title is written (check with r.info) and the problem is in reading it. But it's not terribly important feature, so I am fine with marking this as failing on Windows and move on.

So, I’ve marked the `-f` flag as failing on Windows.

FYI, I confirmed that the title is indeed written using `r.info`, so the issue seems to be with reading it in `g.list`.(see: https://github.com/NishantBansal2003/grass/actions/runs/15772133164/job/44458838955?pr=4#step:12:1020 and https://github.com/NishantBansal2003/grass/actions/runs/15772133164/job/44458838955?pr=4#step:12:1069)
